docs/UserGuide.md

```diff
@@ -249,49 +260,84 @@ Examples of usage:
 * Cannot be used in succession to undo previously executed commands besides the last executed one.
 
+Examples of usage:
+* Using `undo` after an `add` command will remove the recently added student.
+* Using `undo` after an `edit` command will restore the details of the student back to his/her previous state.
+* Using `undo` after a `delete` command will add the recently deleted student.
+
+Scenario for undoing an edit command:
+![UndoingAnEditCommand](images/UndoCommandScenario.png) *Figure 8: Scenario of using an undo command following an edit command.*
+
+Figure *8* depicts a scenario in which an `undo` command is used to restore any previous changes made to a student using the `edit` command. Let's say you want to remove the tags for Jane Doe who is at index 3. 
+The command `edit 3 t/` will remove all tags from Jane Doe. However, you then changed your mind in removing all tags from Jane Doe. You can then use the `undo` command to restore all tags for Jane Doe.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** 
+* After using an `undo` command successfully, you cannot use an `undo` command following that `undo` command.
+* Using an `undo` command after an `edit` command will not restore the student's index back to his/her original index.
+  * If the student's original index is 1, using an `edit` command will place him/her at the last index of Tracey
+  * Using an `undo` command will not restore the student's index back to 1
+</div>
+
 ### Redoing an action
-Reverses an executed undo command.
+Reverse an executed undo command.
 
 Format: `redo`
 * Reverses only the last executed `undo` command.
 * You can only use the `redo` command after executing an `undo` command.
 * Cannot be used in succession to reverse previously executed `undo` commands besides the last executed one.
 
+Examples of usage:
+* Using `redo` after using `undo` on an `add` command will counteract the effect of undoing an `add` command and the overall result adds a student into Tracey.
+* Using `redo` after using `undo` on a `delete` command will counteract the effect of undoing a `delete` command and the overall result deletes the student from Tracey.
+* Using `redo` after using `undo` on an `edit` command will counteract the effect of undoing an `edit` command and the overall result edits the details of the student in Tracey.
+
+Scenario for redoing after undoing an edit command:
+![RedoingTheUndoingAnEditCommand](images/RedoCommandScenario.png) *Figure 9: Scenario of using a redo command after using an undo command following an edit command.*
+
+Following the scenario from *Figure 8*, *Figure 9* depicts the scenario in which you have decided once again to change your mind and remove the tags for Jane Doe. You can use the `redo` command
+to counteract the effects of the `undo` command on the `edit` command, and the overall results in the tags being removed from Jane Doe.
```

Comment:
"and the overall results in the tags being removed from Jane Doe." i think this part you can say "to counteract the effects of the `undo` command. At the end, all tags will be removed from Jane Doe."